runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -479,9 +479,26 @@ TR_J9ServerVM::classHasBeenReplaced(TR_OpaqueClassBlock *clazz)
 bool
 TR_J9ServerVM::classHasBeenExtended(TR_OpaqueClassBlock *clazz)
    {
+   uintptrj_t classDepthAndFlags = 0;
    JITaaS::J9ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   stream->write(JITaaS::J9ServerMessageType::VM_classHasBeenExtended, clazz);
-   return std::get<0>(stream->read<bool>());
+   bool dataFromCache = JITaaSHelpers::getAndCacheRAMClassInfo((J9Class *)clazz, _compInfoPT->getClientData(), stream, JITaaSHelpers::CLASSINFO_CLASS_DEPTH_AND_FLAGS, (void *)&classDepthAndFlags);
+   if(!clazz)
```

Comment:
I would move this test above. There is no point to check the cache if `clazz` is NULL

Reply:
Sounds good, I will do that.